src/v2/providers/pubsub.ts

```diff
@@ -86,29 +103,51 @@ export class Message<T> {
   }
 }
 
-/** The interface published in a Pub/Sub publish subscription. */
+/**
+ * The interface published in a Pub/Sub publish subscription.
+ * @param T - Type representing `Message.data`'s json format
+ */
 export interface MessagePublishedData<T = any> {
+  /**  Google Cloud Pub/Sub message. */
   readonly message: Message<T>;
-  readonly subscription: string;
+  /** A subscription resource. */
+  readonly subscription: PubSubSubscription;
 }
 
 /** PubSubOptions extend EventHandlerOptions but must include a topic. */
 export interface PubSubOptions extends options.EventHandlerOptions {
-  topic: string;
+  /** The Pub/Sub topic to watch for message events */
+  topic: PubSubTopic;
 }
 
-/** Handle a message being published to a Pub/Sub topic. */
+/**
+ * Handle a message being published to a Pub/Sub topic.
+ * @param topic - The Pub/Sub topic to watch for message events.
+ * @param handler - runs every time a Cloud Pub/Sub message is published
+ * @param T - Type representing `Message.data`'s json format
```

Comment:
Suggest "JSON" or `json` if we want the literal.

Reply:
JSON!